cloud/scope/powervs_cluster.go

```diff
@@ -1102,13 +1107,14 @@ func (s *PowerVSClusterScope) createVPC() (*string, error) {
 // ReconcileVPCSubnets reconciles VPC subnet.
 func (s *PowerVSClusterScope) ReconcileVPCSubnets() (bool, error) {
 	subnets := make([]infrav1beta2.Subnet, 0)
+	vpcZones, err := regionUtil.VPCZonesForVPCRegion(*s.VPC().Region)
+	if err != nil {
+		return false, err
+	}
 	// check whether user has set the vpc subnets
 	if len(s.IBMPowerVSCluster.Spec.VPCSubnets) == 0 {
 		// if the user did not set any subnet, we try to create subnet in all the zones.
-		vpcZones, err := regionUtil.VPCZonesForVPCRegion(*s.VPC().Region)
-		if err != nil {
-			return false, err
-		}
+
 		if len(vpcZones) == 0 {
```

Comment:
I think we should move this check right after we fetch teh vpcZones? so we can early return?

Reply:
Sure, I have moved it now